### PR TITLE
ESNext: update esprima to properly parse regex tokens

### PR DESCRIPTION
**Pull request by mikesherov**

### CODE REVIEW

**Comment by mikesherov** [comment]
@markelog or @mdevils or @mrjoelkemp please review and if LGTM'ed, please merge before we release 1.10

**Comment by mdevils** [comment]
LGTM.

What means `strict`?

File: lib/string-checker.js [hunk 1]
```diff
@@ -89,7 +89,15 @@ StringChecker.prototype = {
         var parseError;
 
         try {
-            tree = this._esprima.parse(str, {loc: true, range: true, comment: true, tokens: true});
+            tree = this._esprima.parse(str, {
+                loc: true,
+                range: true,
+                comment: true,
+                tokens: true,
+                // TODO: sourceType is required to parse 'import' and 'export' statements,
```
Comment (markelog) [inline lib/string-checker.js]: Why this is `TODO`, we can change this later?

File: lib/string-checker.js [hunk 2]
```diff
@@ -89,7 +89,15 @@ StringChecker.prototype = {
...
+                // but has the side effect of forcing all code to be considered strict
```
Comment (markelog) [inline lib/string-checker.js]: Is there a relevant link about this?

File: lib/string-checker.js (hunk 1 above)
Comment (mikesherov) [inline lib/string-checker.js]: At a later date, we may consider revising this? I suppose I should just leave an issue, and not put the todo in code. Amending.

**Comment by mikesherov** [comment]
By `strict` I mean strict mode. Because modules, as per spec, are always in strict mode. 

But if a user want to use es6, and not modules, they could theoretically want non strict mode. and then we'd need to provide the option of whether to say module or not as an esprima option, which is annoying and we can save for later until someone complains.

**Comment by markelog** [comment]
Could you mention in the commit what issues do you fix?

**Comment by mikesherov** [comment]
@markelog yes, will do.

**Comment by markelog** [comment]
LGTM

**Pull request closed without merging**